rust/src/lib.rs

```diff
@@ -107,3 +107,6 @@ pub mod plugin;
 pub mod lzma;
 pub mod util;
 pub mod ffi;
+
+//Re-export htp symbols
+pub use htp::c_api::*;
```

Comment:
What is this magic ?

Reply:
This magic makes sure that the rust compiler includes the libhtp c_api symbols in the output `libsuricata_rust` so that suricata links correctly.

Reply:
Could Suricata link correctly by linking to some libhtp.a independently of libsuricata_rust ?

Reply:
@jasonish thoughts here ?
You mentioned that it may be good to have more than one suricata crate for the building process with C and rust intertwined...

Reply:
> @jasonish thoughts here ?

This seems fine.  Suricata would be pulling in the .rlib anyways. This is just for standalone installs of libhtp as C library. Do I have that right @cccs-rtmorti ?

Reply:
@jasonish Yes, the point of this magic here to not rely on libhtp being a separate library to link against in suricata, since it is all wrapped up in libsuricata_rust, and building suricata with a separate libhtp is no longer an option.

For standalone installs of libhtp as a shared library (for some other project, say), suricata shouldn't be picking up the header (which is the other conversation about naming the header `htp_rs.h` or something else non-conflicting), and suricata doesn't / shouldn't care about any installed `libhtp.so` or `libhtp.a` the same way it doesn't care about `libboost.so` (which is what this PR does now - libhtp is just built into libsuricata_rust).